src/main/java/com/microsoft/graph/exceptions/AuthenticationException.java

```diff
@@ -0,0 +1,33 @@
+package com.microsoft.graph.exceptions;
+
+import javax.annotation.Nonnull;
+
+import com.microsoft.graph.core.ClientException;
+
+/**
+ * Exceptions thrown by the authentication provider when the authentication sequence has failed
+ */
+public class AuthenticationException extends ClientException {
```

Comment:
@ddyett following up on your comment from a long time ago. Bringing all the non-gen stuff from the service library allowed me to remove a lot of duplication across the board. Now I wonder whether we actually need an Authentication exception? (note that the client exception wasn't in this repo before)